src/h5web/vis-packs/core/matrix/utils.ts

```diff
@@ -21,3 +23,29 @@ export function getFormatter(
 
   return (val) => (val as string).toString();
 }
+
+export function sliceToCsv(slice: NdArray<Primitive<PrintableType>>): string {
+  let csv = '';
+
+  if (slice.shape.length === 1) {
+    for (let i = 0; i < slice.shape[0]; i++) {
+      csv += `${slice.get(i).toString()}\n`; // complex numbers are stringifyied as two values
+    }
+
+    return csv;
+  }
+
+  if (slice.shape.length === 2) {
+    for (let i = 0; i < slice.shape[0]; i++) {
+      let line = '';
+      for (let j = 0; j < slice.shape[1]; j++) {
+        line += `${slice.get(i, j).toString()},`; // complex numbers are stringifyied as two values
+      }
+      csv += `${line.replace(/,$/u, '')}\n`;
```

Comment:
```suggestion
      csv += `${line.slice(0, -1)}\n`;
```
or
```suggestion
      csv += line.replace(/,$/u, '\n');
```

Reply:
Ha, clever 😄 -- I'll use the second one, as I like the self-documenting aspect of the regex.